File: tests/fixtures/graph-src-tier3/scala/Shapes.scala
package com.example.geo

import scala.math.sqrt

val pi: Double = 3.14159

type Scalar = Double

/** Free function. */
def circleArea(radius: Scalar): Scalar = {
  radius * radius * pi
}

trait Shape {
  def area(): Scalar
}

class Circle(val radius: Scalar) extends Shape {
  def area(): Scalar = {
    circleArea(radius)
  }
}

abstract class Base(val tag: Scalar)

// The primary parent after `extends` (Base) is the only `extends` edge; the
// `with Shape` mixin produces no edge, matching CodeGraph 0.9.7.
class Ring(val r: Scalar) extends Base(r) with Shape {
  def area(): Scalar = r
}

object Registry {
  def count(): Int = 0
}

enum Suit {
  case Hearts
  case Spades
}

def hypot(a: Scalar, b: Scalar): Scalar = sqrt(a * a + b * b)

File: tests/fixtures/graph-src-tier3/php/Shapes.php
<?php

const PI = 3.14159;

/**
 * Free function.
 */
function circleArea(float $radius): float {
    return $radius * $radius * PI;
}

interface Shape {
    public function area(): float;
}

trait Named {
    public function label(): string {
        return "shape";
    }
}

class Circle implements Shape {
    use Named;

    public float $radius;
    const KIND = "circle";

    public function __construct(float $radius) {
        $this->radius = $radius;
    }

    public function area(): float {
        return circleArea($this->radius);
    }
}

enum Suit {
    case Hearts;
    case Spades;
}
